### PR TITLE
Add Spanish stopwords

### LINKED ISSUE #1280: Add stopwords for Spanish
Luis Benítez has sent us an excel sheet with Spanish stop words and the following links:
https://www.ranks.nl/stopwords/spanish
http://googleseo.marketing/lista-de-stop-words-o-palabras-vacias-en-espanol/

We should compose a Spanish stop word list based on these lists and build a language specific implementation of the stopword researches.

### PR DESCRIPTION
**Pull request by IreneStr**
## Summary

This PR can be summarized in the following changelog entry: Add Spanish support for the stopWordsInUrl and stopWordsInKeyword assessment. Props Luis Benítez.

## Relevant technical choices: Adds support for the urlStopWordsAssessment and keywordStopWordsAssessment for Spanish.

* Add Spanish stopwords list
* I've used the same structure as is used for the syllable count/Flesch Reading: a helper function that provides the research with the correct stopwords list (`src/config/stopwords.js`) and json files with word list in config folder (`src/config/stopwords/en.json` and `es.json`)
* Add Spanish support for the keywordStopWordsAssessment and urlStopWordsAssessment

## Test instructions

This PR can be tested by following these steps:

* Link this branch to your local WordPress install. Don't forget to run a `grunt build:js`.
* Set the site language to Spanish. 
* Create a post with a focus keyword containing one or more of the words from `es.json`.
* Put also one of the stopwords in the slug of the post.
* Check whether the following two bullets are shown:
schermafbeelding_2017-11-10_om_14_58_11

* Repeat the above steps for English to check I didn't break anything for English.
* Repeat the above steps for another locale. The above bullet should not show up, even when you are using English/Spanish stop words.

Fixes #1280

### CODE REVIEW

File: src/config/stopwords.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+
+let es = require( "./stopwords/es.json" ).stopwords;
+let en = require( './stopwords/en.json' ).stopwords;
+
+let languages = { es, en };
+
+module.exports = function( locale = "en_US" ) {
+	let language = getLanguage( locale );
+
+	if( languages.hasOwnProperty( language ) ) {
```
Comment (afercia) [inline src/config/stopwords.js]: missing space

File: src/researches/stopWordsInKeyword.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
 /** @module researches/stopWordsInKeyword */
 
-var stopWordsInText = require( "./stopWordsInText.js" );
-
-var escapeRegExp = require( "lodash/escapeRegExp" );
+import stopWordsInText from  "./stopWordsInText.js";
```
Comment (afercia) [inline src/researches/stopWordsInKeyword.js]: double space

**Comment by afercia** [comment]
CR 🚧 

Minor coding standards. Also, in `src/researches/stopWordsInUrl.js` there's still a require for `stopWordsInText` while in other files it has been changed in an import.

**Comment by afercia** [comment]
Testing a bit more, seems there's a problem with accented letters.

In this case, it's returning both `como` and `cómo`

screen shot 2017-11-13 at 10 19 06

an din this case, both `cuando` and `cuándo`

screen shot 2017-11-13 at 10 30 58

**Comment by afercia** [comment]
To my understanding, here's what happens, for example with `como` and `cómo`:

`como`:
the stop words list is processed by `createWordRegex.js` where `replaceDiacritics()` replaces the `ó` in the stop words list with `o`. Then, there are 2 `como` in the list and the entered term matches twice

`cómo`:
actually the processed stop words list doesn't have diacritics (see above) but more importantly regexes in JavaScript take into account what a "word character" is in JavaScript:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
> Note: JavaScript's regular expression engine defines a specific set of characters to be "word" characters. Any character not in that set is considered a word break. This set of characters is fairly limited: it consists solely of the Roman alphabet in both upper- and lower-case, decimal digits, and the underscore character. Accented characters, such as "é" or "ü" are, unfortunately, treated as word breaks.

**Comment by IreneStr** [comment]
@afercia How I understand it, the fact that regexes in javascript have a limited list of word characters, only matters when using wildcards like `.` and `\b`. When I remove `replaceDiacritics()` it does match (for example) 'habré', which is on the list, but not 'habre', which is not on the list. I'll investigate a little further how to break no other regexes.

**Comment by IreneStr** [comment]
@afercia Mystery explained: como, cómo, cuando and cuándo are all on the Spanish list 🙃

**Comment by IreneStr** [comment]
@afercia It turns out this ^ is not the full explanation. 
- 'mi' in the slug triggers both matches for 'mi' and 'mí'. Result: The slug for this page contains stop words, consider removing them. ❌ 
- 'mí' in the slug only triggers a match for 'mí'. Result: The slug for this page contains a stop word, consider removing it. ✅

**Comment by afercia** [comment]
To my understanding, the main issue here is about how regexes work in JavaScript 🙂 
> Accented characters, such as "é" or "ü" are, unfortunately, treated as word breaks.

`createWordRegex.js` uses a regex and is then used by `stopWordsInText.js`. If the string passed to the regex contains a diacritic, for example `adónde` JavaScript will treat the `ó` as a word break so the actual string used to match stop words will be `ad nde` (I guess actually without the space but treated as 2 separated words).

**Comment by IreneStr** [comment]
Closed, because we decided in the recalibration meeting that the StopwordsInUrl and StopWordsInKeyword assessments will be removed soon.

**Pull request closed without merging**